cobc/cobc.h

```diff
@@ -105,9 +105,10 @@ enum cb_format {
 	CB_FORMAT_ICOBOL_CRT,	/* ICOBOL Free-form format (CRT) */
 	CB_FORMAT_ACUTERM,	/* ACU Terminal format, named "TERMINAL" */
 	CB_FORMAT_COBOLX,	/* GCOS's COBOLX */
+	CB_FORMAT_AUTO, 	/* Auto-detect format */
 };
 #define CB_SF_FREE(sf) (sf == CB_FORMAT_FREE)
-#define CB_SF_FIXED(sf) (sf == CB_FORMAT_FIXED || sf == CB_FORMAT_COBOL85)
+#define CB_SF_FIXED(sf) (sf == CB_FORMAT_FIXED || sf == CB_FORMAT_COBOL85 || sf == CB_FORMAT_AUTO)
```

Comment:
We shouldn't reach this macro when the SF is still auto, should we?

Reply:
Actually, we reach it: if you compile in listing mode, autodetection will detect that the file is fixed, but then it will switch it back to `CB_FORMAT_AUTO` at the end of preprocessing, so that, when you reach the listing phase, either you have to redo autodetection or just assume it was fixed.

Reply:
I see, then the missing piece is to set `source_format` in the `struct list_files` after ppopen.

Reply:
Indeed, I didn't notice it was saved there. I modified `ppopen` to update it and display the header afterwards.